usr/src/lib/libzfs/common/libzfs_mount.c

```diff
@@ -1166,6 +1176,14 @@ mountpoint_cmp(const void *arga, const void *argb)
 	const char *a = mounta;
 	const char *b = mountb;
 	boolean_t gota, gotb;
+	uint64_t zoneda, zonedb;
+
+	zoneda = zfs_prop_get_int(za, ZFS_PROP_ZONED);
+	zonedb = zfs_prop_get_int(zb, ZFS_PROP_ZONED);
+	if (zoneda && !zonedb)
+		return 1;
+	if (!zoneda && zonedb)
```

Comment:
Should there be a case for both datasets being zoned? (so don't mount in GZ, should not happen i NGZ)

Reply:
This is just the sort function used from qsort() so we only need to prefer non-zoned to zoned and the zoned will end up at the bottom of the list.